src/utilities/BatchTable.js

```diff
@@ -0,0 +1,60 @@
+import { BatchTableHierarchyExtension } from './BatchTableHierarchyExtension.js';
+import { FeatureTable } from './FeatureTable.js';
+
+export class BatchTable extends FeatureTable {
+
+	constructor( buffer, batchSize, start, headerLength, binLength ) {
+
+		super( buffer, start, headerLength, binLength );
+		this.batchSize = batchSize;
+
+		const extensions = this.header.extensions;
+		if ( extensions ) {
+
+			this.extensions = {};
+
+			if ( extensions[ '3DTILES_batch_table_hierarchy' ] ) {
+
+				this.extensions[ '3DTILES_batch_table_hierarchy' ] = new BatchTableHierarchyExtension( this );
+
+			}
+
+		}
+
+	}
+
+	getData( key, componentType = null, type = null ) {
+
+		return super.getData( key, this.batchSize, componentType, type );
+
+	}
+
+	getDataForId( id, target = {} ) {
+
+		if ( id < 0 || id >= this.batchSize ) {
+
+			throw new Error( `BatchTable: id value "${ id }" out of bounds for "${ this.batchSize }" features number.` );
+
+		}
+
+		for ( const key of this.getKeys() ) {
+
+			if ( key !== 'extensions' ) {
+
+				target[ key ] = this.getData( key )[ id ];
+
+			}
+
+		}
+
+		for ( const extension in this.extensions ) {
+
+			target[ extension ] = this.extensions[ extension ].getDataForId( id );
+
+		}
```

Comment:
What's the rationale here? Is it the case that _every_ extension will have `getDataForId`?

Reply:
Whoops! Indeed adding a check for `getDataForId` existence is a bug prevention I completely missed. I'll add that.

I know of a research lab that uses custom made 3D tiles extensions, but I don't know if they use some on batch tables (perhaps @jailln knows more of it). Anyway, `BatchTable.extensions` would need to be public, with an extension registration tool for `BatchTable` to allow adding custom extensions. So I agree that we will probably only deal with batch table hierarchy extension.